src/wp-includes/block-template-utils.php

```diff
@@ -36,22 +36,9 @@
  *     @type string $wp_template_part Theme-relative directory name for block template parts.
  * }
  */
-function get_block_theme_folders( $theme_stylesheet = null ) {
-	$theme_name = null === $theme_stylesheet ? get_stylesheet() : $theme_stylesheet;
-	$root_dir   = get_theme_root( $theme_name );
-	$theme_dir  = "$root_dir/$theme_name";
-
-	if ( file_exists( $theme_dir . '/block-templates' ) || file_exists( $theme_dir . '/block-template-parts' ) ) {
-		return array(
-			'wp_template'      => 'block-templates',
-			'wp_template_part' => 'block-template-parts',
-		);
-	}
-
-	return array(
-		'wp_template'      => 'templates',
-		'wp_template_part' => 'parts',
-	);
+function get_block_theme_folders( $theme_stylesheet = '' ) {
+	$theme = wp_get_theme( $theme_stylesheet );
```

Comment:
Related to @costdev's comment in https://github.com/WordPress/wordpress-develop/pull/5202/files#r1325669765, we should continue to expect `null` here. No point in changing the default. The `wp_get_theme()` function expects a string, and we can easily cast that.
```suggestion
function get_block_theme_folders( $theme_stylesheet = null ) {
	$theme = wp_get_theme( (string) $theme_stylesheet );
```

Reply:
@felixarntz I have restored back `null` but I feel type casting to string is not needed here. 
`wp_get_theme` does have an empty check almost immediately which would handle `null`. Should we still type cast just to respect function definition?

Reply:
@kt-12 Yes, personally I would say we should try to respect the expected signature since it's trivial enough.

Reply:
@felixarntz It was indeed a good suggestion, I was tracing forward a bit to see if there is a chance of possible failure and I did get an instance where the function would have worked differently had we not typecasted to string.